docs/demo.md

```diff
@@ -0,0 +1,218 @@
+# Garden demos
+
+Garden leverages [Storybook](https://storybook.js.org/) for generating component
+demos with story controls. Run `yarn start` to build and serve package demos in
+development mode (with hot reloading).
+
+The development goal for demos is to render components in isolation with the
+least amount of structural scaffolding together with the greatest amount of
+control for content and flexibility. Along with global toolbar items (preview
+size, locale direction, CSS Bedrock, etc), controls should stress every aspect
+of a component's layout and behavior.
+
+## Best practice
+
+The following list of dos and don'ts outline demo code expectations.
+
+### Do this
+
+- Expose controls for all essentials – inherent props, conditional subcomponent
+  renders, child content
+- Allow natural component layout – full width for block, partial width for inline
+- Use the Garden `Grid` to manage layout for components that respond to
+  placement values (start, end, top, bottom, etc)
+- Rely on `useArgs` from Storybook's client API in order to keep component state
+  connected with it's associated Storybook control
```

Comment:
This is `true` (ha, see what I did there). But this is one where I prefer to go with the grain of Storybook and see that boolean props are `undefined` as expected. Taking this approach lowers the devX bar (I don't have to chase down pre-defining all the component boolean `args`) and it helped us find inconsistent prop default definitions (https://github.com/zendeskgarden/react-components/pull/1265#discussion_r775973782).

Reply:
Hmm, I disagree with this to some degree, since a boolean for example is a primitive data type that would default to false in most cases, and can mitigate inconsistencies with types. Definitely not a blocker.

Reply:
I agree that Garden could've opted for `false` defaults, and I could even agree that might be ideal. But the decision to go with `undefined` booleans throughout the component API long pre-dates the making of this PR.